bindings.go

```diff
@@ -388,3 +390,141 @@ func (n Node) PrevNamedSibling() *Node {
 func (n Node) Edit(i EditInput) {
 	C.ts_node_edit(&n.c, i.c())
 }
+
+// QueryErrorType - value that indicates the type of QueryError.
+type QueryErrorType int
+const (
+	QueryErrorNone QueryErrorType = iota
+	QueryErrorSyntax
+	QueryErrorNodeType
+	QueryErrorField
+	QueryErrorCapture
+)
+
+// QueryError - if there is an error in the query,
+// then the Offset argument will be set to the byte offset of the error,
+// and the Type argument will be set to a value that indicates the type of error.
+type QueryError struct {
+	Offset uint32
+	Type QueryErrorType
+}
+func (qe *QueryError) Error() string {
+	switch qe.Type {
+	case QueryErrorNone:
+		return ""
+
+	case QueryErrorSyntax:
+		return fmt.Sprintf("syntax error (offset: %d)", qe.Offset)
```

Comment:
Creating a `type SyntaxError struct{ Offset uint32 }` may be a good idea. Same for other the error kinds. Can be in the next PR, though.

Reply:
Ok, makes sense, but if we replace `QueryError` (which contains type and offset) by many errors (per type) which will contain just an offset.
Otherwise we'll not get rid of this switch just moving Sprintfs under each `Error` struct.

Is it what you suggest? To make a struct per error type and return just an error and force people to get type and offset by _switch type-cast_?

I tried also to keep API similar to the C-original one, so users will not have to think to much how to use it (knowing the original names).